### PR TITLE
Add generateModelURL utility function

### LINKED ISSUE #27: Create utility method for generating model URLs

### PR DESCRIPTION
**Pull request by hatched**
Fixes #27

### CODE REVIEW

File: api/test-client.js [hunk 1]
```diff
@@ -588,3 +587,13 @@ tap.test('connectAndLogin', t => {
 
   t.end();
 });
+
+tap.test('generateModelURL', t => {
+  t.test('returns valid url', t => {
```
[the remark this answers is not in the document]
Comment (hatched) [reply api/test-client.js]: The idea did cross my mind but I wasn't exactly sure how resilient we need to be and I figured that maybe the docstrings would be sufficient.

File: Makefile [hunk 2]
```diff
@@ -49,6 +49,7 @@ help:
 	@echo "make generate - generate API in api/facades and docs from schema"
 	@echo "make update-schema - update schema with current juju develop"
 	@echo "make test - run unit tests"
+	@echo "make test-js - run js unit tests"
```
Comment (frankban) [inline Makefile]: Thanks! Also add lint-js while you are there?

File: api/client.js [hunk 3]
```diff
@@ -182,6 +182,19 @@ function connectAndLogin(url, credentials, options, callback) {
 }
 
 
+/**
+  Returns a url that is to be used to connect to a supplied model uuid on the
```
Comment (frankban) [inline api/client.js]: s/url/URL/ in this line and the following.

File: api/client.js [hunk 4]
```diff
@@ -182,6 +182,19 @@ function connectAndLogin(url, credentials, options, callback) {
...
+  supplied controller url.
+  @param {String} controllerURL The url that's used to connect to the controller.
```
Comment (frankban) [inline api/client.js]: Looking at the code, this does not seem technically a URL. Maybe `controllerHost` is more adequate?

File: api/client.js [hunk 5]
```diff
@@ -182,6 +182,19 @@ function connectAndLogin(url, credentials, options, callback) {
...
+    The `connect` method handles redirections so the public URL is fine.
+  @param {String} modelUUID The UUID of the model to connect to.
+  @returns {String} The fully qualified wss url to connect to the model.
```
Comment (frankban) [inline api/client.js]: s/url/URL/

File: api/client.js [hunk 6]
```diff
@@ -182,6 +182,19 @@ function connectAndLogin(url, credentials, options, callback) {
 }
 
 
+/**
+  Returns a url that is to be used to connect to a supplied model uuid on the
+  supplied controller url.
+  @param {String} controllerURL The url that's used to connect to the controller.
+    The `connect` method handles redirections so the public URL is fine.
```
Comment (frankban) [inline api/client.js]: This sentence seems weird without a bit of context. Also the connect method does not handle redirections, connectAndLogin does.

File: Makefile (hunk 2 above)
Comment (hatched) [reply Makefile]: done

File: api/client.js (hunk 3 above)
Comment (hatched) [reply api/client.js]: done

File: api/client.js (hunk 4 above)
Comment (hatched) [reply api/client.js]: Yeah I was thinking the same thing :) Updated

File: api/client.js (hunk 6 above)
Comment (hatched) [reply api/client.js]: updated

File: api/client.js (hunk 5 above)
Comment (hatched) [reply api/client.js]: done

**Pull request merged**